### ISSUE TITLE
Running Vite fails with `ERR_CHILD_PROCESS_STDIO_MAXBUFFER` on very large `public` directory

### ISSUE THREAD

**Issue by mz8i** [vikejs/vike#1982]
### Description

I have a very large number of files in my `public` directory (>16k) - the project is a pre-rendered data visualisation web app with numerous small CSV files to be loaded from a static hosting during usage.

After running `vite` in my project directory, I get a Bun/Node-related error due to `ERR_CHILD_PROCESS_STDIO_MAXBUFFER`. A similar error occurs when running `vite build`.

```
bun run dev
$ vite
3:13:13 PM [vike][config][Error] An error was thrown:
RangeError: stdout maxBuffer length exceeded
    at Socket.onChildStdout (node:child_process:487:14)
    at Socket.emit (node:events:518:28)
    at addChunk (node:internal/streams/readable:561:12)
    at readableAddChunkPushByteMode (node:internal/streams/readable:512:3)
    at Socket.Readable.push (node:internal/streams/readable:392:5)
    at Pipe.onStreamRead (node:internal/stream_base_commons:189:23) {
  code: 'ERR_CHILD_PROCESS_STDIO_MAXBUFFER',
  cmd: 'git -c core.quotepath=off ls-files --exclude="**/node_modules/**" --exclude="**/*.telefunc.*" --exclude="dist/**" --others --exclude-standard --cached --stage',
  stdout: '.vscode/launch.json\n' +
    'public/data/fact-tables/Transport1B/34432.csv\n' +
[MORE LINES - TRUNCATED]
    'public/data/fact-tables/Transport1B/44211.csv\n' +
    'public/data/fact-tables/Transport1B/44212.cs'... 1038576 more characters,
  stderr: ''
}
```

It seems that in both Bun and Node, the default `maxBuffer` for `child_process.exec` is `1024 * 1024` and `vike` does not explicitly pass `maxBuffer` so this will fail for a large number of files and/or long paths?

**Comment by brillout** [comment]
Have you tried `NODE_OPTIONS`? Closing but I'll re-open if it turns out to be somewhat related to Vike.

**Comment by mz8i** [comment]
I've done a bit of research on this, and it's possible I missed something, but so far I don't think this is possible to solve with a command line option but rather needs to be handled in the library code (see SO link at the end of this comment). But maybe you have some particular option in mind that you think I should pass through `NODE_OPTIONS`?

This is specific to the `child_process` module - see the docs for `exec`, and specifically the `maxBuffer` option: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback

Vike runs `git ls-files` through child_process, and from what I saw in the source code, doesn't pass a maxBuffer option - which means that if the output from `git ls-files` is larger than the default 1024 * 1024 bytes, the process will error.

More discussion of the nature of the problem: https://stackoverflow.com/a/63304039

**Comment by mz8i** [comment]
Judging by the error output, the problem arises when running [this line](https://github.com/vikejs/vike/blob/main/vike%2Fnode%2Fplugin%2Fplugins%2FimportUserCode%2Fv1-design%2FgetVikeConfig%2FcrawlPlusFiles.ts#L348), and based on my basic research so far, perhaps it would be enough to pass a `maxBuffer: Infinity` option in [this function](https://github.com/vikejs/vike/blob/main/vike%2Fnode%2Fplugin%2Fplugins%2FimportUserCode%2Fv1-design%2FgetVikeConfig%2FcrawlPlusFiles.ts#L348)

**Comment by brillout** [comment]
:+1: Makes sense, let's try this.

**Comment by brillout** [comment]
Fix pre-released as `0.4.204-commit-386d2d8`.

**Comment by mz8i** [comment]
Thanks a lot! I tried to test it but I ran into https://github.com/vikejs/vike/issues/1986 (I was on `0.4.199` before and that new issue appears on every version starting from `0.4.200`, including the pre-release above). I can test again once https://github.com/vikejs/vike/issues/1986 is resolved

**Comment by brillout** [comment]
> I ran into #1986

[Fix](https://github.com/brillout/vite-plugin-server-entry/commit/36d6cbba88f155b91d9dd57c16d7fa8b69a72241) released in `@brillout/vite-plugin-server-entry@0.5.3`. Update your lock file.

**Comment by mz8i** [comment]
I can confirm the issue described here is fixed. Thanks again!
